packages/wallet/src/types.ts

```diff
@@ -102,3 +103,9 @@ export interface ObservableWallet {
   submitTx(tx: Cardano.NewTxAlonzo): Promise<void>;
   shutdown(): void;
 }
+
+export interface GovernanceApi {
+  getVotingKey(derivationPath: number[]): Promise<Ed25519KeyPair>;
+  buildDelegation(props: BuildDelegationProps): Promise<Cardano.AuxiliaryData>;
+  signDelegation(props: SignDelegationProps): Promise<Cardano.Ed25519Signature>;
```

Comment:
Shouldn't this take in the same thing that `buildDelegation` returns?

```suggestion
  signDelegation(props: Cardano.AuxiliaryData): Promise<Cardano.Ed25519Signature>;
```

Then signDelegation implementation can just use `coreToCsl.txMetadatum` to serialize it